src/ice_usb.c

```diff
@@ -36,6 +37,8 @@
 #include "tinyuf2/uf2.h"
 #include "tinyuf2/board_api.h"
 
+#define WATCHDOG_DELAY 3000
```

Comment:
Is that 3s? Is it not a bit short?
Flash programming takes more time, and a slow host could make the transfer arbitrarily long.

Reply:
The tud_dfu_download_cb callback invokes watchdog_update(), which resets the watchdog timer each time. So it should only fire if tud_dfu_download_cb is not invoked again within 3 secs, which I think should be okay even for flash.

Reply:
I missed that somehow, perfect then! 3s of idle between two operations suggests something got stuck, sane to reboot anyway.